HHS/Head-Start-TTADP
# [Prod] Remove null transactions from certain hooks, refactor objective collaborator hook to wait for results before committing

## Description of change
- Certain hooks allowed for "null" to be passed as the transaction option. It's possible that doing so would trigger an additional connection request between sequelize and our database, using one of our limited connection pool and triggering the error that led to the outage.
- A certain hook (objective collaborators, being assigned when reports were approved) needed to be refactored as it was sometimes not waiting for all results to return before committing the transaction, which will throw an error and roll back all the operations.

## How to test
- To confirm that the null transactions aren't an issue any longer, you can observe the transaction logged in the backend logs or in the database in the audit log table.
- Confirm that the case from TTAHUB-3172 is resolved.


## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3138
* https://jira.acf.gov/browse/TTAHUB-3172


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
